core/node/protocol/helpers.go

```diff
@@ -10,3 +17,21 @@ func (e *StreamEvent) GetStreamSettings() *StreamSettings {
 	}
 	return i.GetSettings()
 }
+
+// Scan implements the pgx custom type interface.
+func (m *Miniblock) Scan(src any) error {
+	if src == nil {
+		return nil
+	}
+
+	switch src := src.(type) {
+	case []byte:
+		// Assume the data is serialized protobuf bytes
+		return proto.Unmarshal(src, m)
+	case string:
```

Comment:
I'm slightly worried about implementing the `sql.Scanner` interface in a way that it supports multiple serialisation formats. This interface is used by database packages internally and can cause unexpected and hard to find problems. For example, if the result of `json.Marshal(miniblock)` is written it can't be read from the database because it tries to parse it as protobuf. If we implement the scan function we should also implement `database.Value` and be precise and pass miniblocks instead of byte slices.

For now I personally prefer that we keep the current convention and call `proto.Marshal` explicit as we do in other places and make a new issue to implement the scanner and value interfaces and migrate to passing miniblocks instead of byte slices.

Reply:
Agree, removed this part